db.go

```diff
@@ -1721,6 +1760,75 @@ func (db *DB) filterPrefixesToDrop(prefixes [][]byte) ([][]byte, error) {
 	return filtered, nil
 }
 
+func (lk *lockedKeys) update(fn func(keys map[uint64]struct{})) {
+	lk.Lock()
+	defer lk.Unlock()
+	fn(lk.keys)
+}
+
+func (lk *lockedKeys) view(fn func(keys map[uint64]struct{})) {
+	lk.RLock()
+	defer lk.RUnlock()
+	fn(lk.keys)
+}
+
+func (db *DB) isBannedHelper(key uint64) bool {
+	banned := false
+	db.bannedPrefixes.view(func(keys map[uint64]struct{}) {
+		if _, ok := keys[key]; ok {
+			banned = true
+		}
+	})
+	return banned
+}
+
+// Returns true, if the key is prefixed with any banned prefix.
+func (db *DB) isBanned(key []byte) bool {
```

Comment:
The `isBannedHelper` can be part of this function. you don't need two functions.

Reply:
Done. Thanks.